Automattic/jetpack
# Track user clicks on Links/Buttons in disconnect modal footer

#### Changes proposed in this Pull Request:

* user should see no difference
* when clicking any link or button in the footer of the disconnect dialog - an event should be recorded so we can see how users are interacting with the modal options (see test plan for details)

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
We are tracking some more clicks but no major change to how or what we track.

#### Testing instructions:

You can see if tracking events are being triggered by looking for the tracking pixels in the Network and checking query parameters - filter network traffic by `t.gif`. Or enabling logging in the browser console, as detailed here PCYsg-nSS-p2. Or visit the internal live tracker website.

* with Jetpack development running locally
* open /wp-admin
* navigate to -> Jetpack -> Dashboard
* click 'Manage Site Connection'
* click the `Jetpack connection` link and check you see the `jetpack_disconnect_dialog_click_learn_about` event
* click the `contact Jetpack support` link and check you see the `jetpack_disconnect_dialog_click_support` event
* click the `Stay Connected` button and check the modal still closes and you see the `jetpack_disconnect_dialog_click_stay_connected` event
* reopen the modal, and click the `Disconnect` button and check you still disconnect and see the `jetpack_disconnect_dialog_click_disconnect` event

Repeat the above from the -> Plugins : Jetpack : `Disconnect and Deactivate` - the events are the same but the `context` of the tracking should be `plugins` rather than `jetpack`

For reference, the links and buttons are illustrated here:
![disconnect_modal](https://user-images.githubusercontent.com/8742105/151022993-540f369f-d276-4aeb-bba3-44794fd68f4e.png)

#### Unit Tests

Unit Tests have been updated and can be checked by running `pnpm t` from the `projects/js-packages/connection/` folder.
